### DISCUSSION TITLE
Hide/show columns

### DISCUSSION THREAD

**Discussion by danisal** [nick-keller/react-datasheet-grid#80] [Q&A] [answered]
Hey @nick-keller , first of all, congratulations on the great library!

And second I saw that it is still in development of new features and I would like to ask if there is a way (or plan) to hide/show columns?  Like when we click on the column name for the context menu and show an option to hide a particular column?

Thanks in advance!

**Comment by nick-keller** [comment, marked as answer]
Hey @danisal and thx for the kind words.

Good news, this is already feasible!
Bad news, it requires a little bit of work on your side.

- First, make sure to use the [dynamic version of DSG](https://react-datasheet-grid.netlify.app/docs/performance/static-vs-dynamic) to be able to add / remove columns on the fly.
- Then you'll need to have a state to hold a Set (or Array) of hidden columns. You could use their ids or name.
- Based on that Set of hidden columns, build the list of visible column objects to pass to DSG (use a `useMemo` for performance)

The way you allow the user to toggle columns is up to you, it could be a menu to the side, or you could use the context menu that is [entirely customizable](https://react-datasheet-grid.netlify.app/docs/api-reference/props#contextmenucomponent).

**Reply by danisal** [reply]
This is great, thanks for the feedback! Manage to implement it! Cheers 🥂
